lib/registryFromSparse.nix

```
@@ -0,0 +1,23 @@
+{ registryFromDownloadUrl }:
+
+{ indexUrl
+, sha256
+, fetchurlExtraArgs ? { }
+}:
+
+let
+  configContents = builtins.readFile "${builtins.fetchurl {
+    inherit sha256;
+    url = "${indexUrl}config.json";
```

```suggestion
    url = "${indexUrl}/config.json";
```

Nit: I'd prefer we include the `/` here rather than assume the user will specify a url with a trailing slash

---

I did that - here, I remove the trailing slash for getting the config json, and additionally i handle it in the 'fetch from download url' case.